### PR TITLE
Create attendance crud UI

### PR DESCRIPTION
**Pull request by MarcuzAizen**
please review, thanks

### CODE REVIEW

**Comment by kertnik05** [comment]
@MarcuzAizen  redesign the ui for the attendance crud. I think it should include the section, grade_level, track, student & subject sched. Imagine what would be records in a class record of the teacher.

**Review by kertnik05 — CHANGES_REQUESTED** [review]
@MarcuzAizen  redesign the ui for the attendance crud. I think it should include the section, grade_level, track, student & subject sched. Imagine what would be records in a class record of the teacher.

**Comment by MarcuzAizen** [comment]
Screenshot from 2021-01-15 08-57-36
Screenshot from 2021-01-15 08-58-02
Screenshot from 2021-01-15 08-57-46

File: app/Http/Controllers/API/AttendanceController.php [hunk 1]
```diff
@@ -48,4 +50,25 @@ public function destroy(Attendance $attendance)
...
     }
+
+    public function getTeachers()
+    {
+        $teachers = User::whereHas('role', function(Builder $query) {
+            $query->where('name', 'teacher');
+        })->latest()->paginate(10);
+
+        return UserResource::collection($teachers);
+    }
```
Comment (kertnik05) [inline app/Http/Controllers/API/AttendanceController.php:61]: For what purpose is this function?

File: app/Http/Controllers/API/AttendanceController.php [hunk 2]
```diff
@@ -48,4 +50,25 @@ public function destroy(Attendance $attendance)
...
+
+    public function getTeacherAttendance(Request $request)
+    {
+        return $request->user()->load([
+
+            'schedule.prospectus.subject',
+            'schedule.section.enrolls.student',
+            'schedule.section.track',
+            'schedule.section'
```
Comment (kertnik05) [inline app/Http/Controllers/API/AttendanceController.php:70]: `Call to undefined relationship [schedule] on model [App\Models\User]`. Relationship from user to schedule model is not `schedule` but `schedules`. Change to `schedules`

File: resources/js/components/teacher/AttendanceModal.vue [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                this.$emit('reload-attendances');
+                $('#attendance-modal').modal('hide');
+                Toast.fire({
+                    icon: 'success',
+                    title: 'Attendance added'
+                });
+                this.$Progress.finish();
+                this.form.reset();
+            }).catch(() => this.$Progress.fail());
+        },
```
Comment (kertnik05) [inline resources/js/components/teacher/AttendanceModal.vue:73]: add new line here to have some space between function for better readability

File: resources/js/pages/teacher/Attendance.vue [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+    created() {
+        this.loadingText = 'Loading...';
+        this.$Progress.start();
+        this.loadAttendances();
+        this.$on('reload-attendances', () => this.loadAttendances());
+    },
+     mounted() {
+        this.searchAttendance = _.debounce(this.searchAttendance, 300);
+    },
```
Comment (kertnik05) [inline resources/js/pages/teacher/Attendance.vue:123]: if you don't have a search function implemented, you can remove this lines and lodash

File: resources/js/pages/teacher/Attendance.vue [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+                this.$Progress.finish();
+            }).catch(() => {
+                this.$Progress.fail();
+            });
+        },
+        sortFunc: function (){
+            return this.attendances.data.schedule.sort(function(a, b){
+                return (a.name > b.name) ? 1 : -1;
+            });
+        },
```
Comment (kertnik05) [inline resources/js/pages/teacher/Attendance.vue:139]: for what purpose is this `sortFunc()`? I don't  see this being implemented in this component.

File: resources/js/pages/teacher/Attendance.vue [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+        showAddAttendanceModal(attendance) {
+            this.editMode = false;
+             this.form.reset();
+            this.form.clear();
+            this.form.fill(attendance);
+            $('#attendance-modal').modal('show');
+        },
```
Comment (kertnik05) [inline resources/js/pages/teacher/Attendance.vue:147]: add new line here to have some space between function for better readability

**Review by kertnik05 — CHANGES_REQUESTED** [review]

**Comment by kertnik05** [comment]
Also please write appropriate and good commit message

**Comment by MarcuzAizen** [comment]
Screenshot from 2021-01-27 10-47-05

**Pull request closed without merging**